### PR TITLE
Auto generate blocks

### PR DESCRIPTION
**Pull request by sstanculeanu**
Implemented the auto generation of the blocks feature, which can be enabled via `auto-generate-blocks` flag. With this flag on, the binary will generate blocks at the speed of the provided flag `block-time-in-milliseconds`

### CODE REVIEW

File: cmd/chainsimulator/config/config.toml [hunk 1]
```diff
@@ -23,3 +23,8 @@
         log-file-life-span-in-sec = 432000 # 5 days
         log-file-prefix = "chain-simulator"
         logs-path = "logs"
+    [config.blocks-generator]
+        # auto-generate-blocks specifies if the chain simulator should auto generate blocks
+        auto-generate-blocks = false
+        #  block-time-in-milliseconds specifies the time between blocks generation in case auto-generate-blocks is enabled
```
Comment (miiu96) [inline cmd/chainsimulator/config/config.toml:29]: remove one space

**Comment by miiu96** [comment]
also can you please add a PR description

File: cmd/chainsimulator/flags.go [hunk 2]
```diff
@@ -96,6 +96,15 @@ var (
 		Usage: "This flag is used to specify the initial epoch when chain simulator will start",
 		Value: 0,
 	}
+	autoGenerateBlocks = cli.BoolFlag{
+		Name:  "auto-generate-blocks",
+		Usage: "Boolean option to specify that blocks should be generated automatically, after a given period of time",
```
Comment (ssd04) [inline cmd/chainsimulator/flags.go:101]: is the second part of the comment valid? "after a given period of time"? or it is referring that there is time between blocks?

File: cmd/chainsimulator/config/config.toml [hunk 3]
```diff
@@ -23,3 +23,8 @@
         log-file-life-span-in-sec = 432000 # 5 days
         log-file-prefix = "chain-simulator"
         logs-path = "logs"
+    [config.blocks-generator]
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply cmd/chainsimulator/config/config.toml:26]: done

File: pkg/process/blocksGenerator.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+	go instance.generateBlocks(ctx)
+
+	return instance, nil
+}
+
+func (generator *blocksGenerator) generateBlocks(ctx context.Context) {
+	log.Info("Running in auto-generate-blocks mode, starting the go routine...")
+
+	timerBlock := time.NewTimer(generator.blockTime)
```
Comment (iulianpascalau) [inline pkg/process/blocksGenerator.go:50]: we can use a time.Ticker instead of timer. In this way, the Reset on L52 is not necessary.

**Review by iulianpascalau — APPROVED** [review]

File: pkg/process/blocksGenerator.go (hunk 4 above)
Comment (sstanculeanu) [reply pkg/process/blocksGenerator.go]: added

**Review by miiu96 — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**